### PR TITLE
Add new options for validation of database connections

### PR DESCRIPTION
**Pull request by basdebakker**
Adds text for new connection pool options. Once ready, the same text may be added for Mendix 9, where the options are supported since 9.24.16.

Still needs review by data storage team.

### CODE REVIEW

File: content/en/docs/refguide/runtime/custom-settings/_index.md [hunk 1]
```diff
@@ -144,6 +144,10 @@ These settings are configured *per runtime instance*. If you have [scaled your a
...
+| <a id="ConnectionPoolingTestOnCreate" href="#ConnectionPoolingTestOnCreate">ConnectionPoolingTestOnCreate</a> | If `true`, a database connection will be validated by the connection pool after being created. Supported since Mendix 10.6.4 and 10.8.0. | false |
+| <a id="ConnectionPoolingTestOnReturn" href="#ConnectionPoolingTestOnReturn">ConnectionPoolingTestOnReturn</a> | If `true`, a database connection will be validated by the connection pool when it is returned to the pool by the application. Supported since Mendix 10.6.4 and 10.8.0. | false |
+| <a id="ConnectionPoolingTestWhileIdle" href="#ConnectionPoolingTestWhileIdle">ConnectionPoolingTestWhileIdle</a> | If `true`, a database connection will be validated when the idle object evictor runs. Supported since Mendix 10.6.4 and 10.8.0. | false |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/custom-settings/_index.md]: You say "a database connection" here. Is there just one database connection which is related to the idle object evictor or could there be several objects with different database connections which are all processed when the idle object evictor runs?
If more than one connection could be validated, then we should say "database connections" (or "all database connections" if they are all validated and not just the ones with idle objects).

File: content/en/docs/refguide/runtime/custom-settings/_index.md [hunk 2]
```diff
@@ -144,6 +144,10 @@ These settings are configured *per runtime instance*. If you have [scaled your a
...
 | <a id="ConnectionPoolingNumTestsPerEvictionRun" href="#ConnectionPoolingNumTestsPerEvictionRun">ConnectionPoolingNumTestsPerEvictionRun</a> | Sets the maximum number of objects to examine during each run of the idle object evictor thread (if any). When a negative value is supplied, `ceil(getNumIdle())/ abs(getNumTestsPerEvictionRun())` tests will be run. This means that when the value is -n, roughly one nth of the idle objects will be tested per run. | -3 |
+| <a id="ConnectionPoolingTestOnBorrow" href="#ConnectionPoolingTestOnBorrow">ConnectionPoolingTestOnBorrow</a> | If `true`, a database connection will be validated by the connection pool before returning it to the application. This can be useful in situations where database connections can be closed by processes outside of the Mendix runtime, like firewalls. Supported since Mendix 10.6.4 and 10.8.0. | false |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/custom-settings/_index.md]: Specifying the versions is difficult here. I would probably go for,

> Supported in Mendix version 10.6 in patch versions 10.6.4 and above, and in Mendix version 10.8.0 and above.

A bit complicated but hopefully, clear enough?

File: content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md [hunk 3]
```diff
@@ -114,6 +114,23 @@ However, if all of the following are true, you should increase the `ConnectionPo
 
 In general, we see that increasing the `ConnectionPoolingMaxActive` value to a (much) higher number is very rarely the right action to take, even if it is unfortunately the action usually taken when you run into connection pooling issues.
 
+### 4.2 Validating database connections
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md]:
```suggestion
### 4.2 Validating Database Connections
```

File: content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md [hunk 4]
```diff
@@ -114,6 +114,23 @@ However, if all of the following are true, you should increase the `ConnectionPo
...
+
+The settings mentioned below are supported since Mendix 10.6.4 and 10.8.0.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md]:
```suggestion
The settings mentioned below are supported in Mendix version 10.6 in patch versions 10.6.4 and above, and in Mendix version 10.8.0 and above.
```

File: content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md [hunk 5]
```diff
@@ -114,6 +114,23 @@ However, if all of the following are true, you should increase the `ConnectionPo
...
+
+In some deployments, database connections can be closed by the network infrastructure, for example some firewalls may close connections when they have been inactive for a long time.
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/custom-settings/tricky-custom-runtime-settings.md]:
```suggestion
In some deployments, database connections can be closed by the network infrastructure, for example by a firewall when they have been inactive for a long time.
```

**Review by MarkvanMents — COMMENTED** [review]
Hi @basdebakker 
Thanks for the PR. Looks good. I have a couple of questions/suggestions below.

File: content/en/docs/refguide/runtime/custom-settings/_index.md (hunk 1 above)
Comment (basdebakker) [reply content/en/docs/refguide/runtime/custom-settings/_index.md]: Too much copy and paste on my end, I suppose. I think "all idle database connections will ..." would be best.

Comment (MarkvanMents) [reply content/en/docs/refguide/runtime/custom-settings/_index.md]: Applied in 9fe39047c28a07989a44016f47f84569c65b6231

File: content/en/docs/refguide/runtime/custom-settings/_index.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/refguide/runtime/custom-settings/_index.md:147]: Applied in 9fe39047c28a07989a44016f47f84569c65b6231

Comment (MarkvanMents) [reply content/en/docs/refguide/runtime/custom-settings/_index.md]: 9fe39047c28a07989a44016f47f84569c65b6231

**Comment by dahfjkg** [comment]
Passed internal technical verification

**Review by MarkvanMents — APPROVED** [review]
Proofread and technically reviewed.
Will be published with 10.8

**Comment by MarkvanMents** [comment]
Merging before 10.8 as it is already released for 10.6.4

**Pull request merged**